tests/test_dock_shortcut_routing_guard.py

```
@@ -0,0 +1,210 @@
+    for cases, disposition in re.findall(
+        r"case\s+(.*?)\s*:\s*\.(dockScoped|focusResolved|mainContainer)",
+        property_body,
+        flags=re.DOTALL,
+    ):
+        result.setdefault(disposition, set()).update(
+            re.findall(r"\.([A-Za-z][A-Za-z0-9_]*)", cases)
+        )
+    return result
+
+
+def balanced_call_bodies(source: str, call_name: str) -> list[str]:
```

Ruff (0.16.0)

[warning] Too many branches (18 > 12)

(PLR0912)